src/openrct2/actions/GameActionRegistry.cpp

```diff
@@ -0,0 +1,399 @@
+                return "SetResearchFunding";
+            case GameCommand::PlaceTrackDesign:
+                return "PlaceTrackDesign";
+            case GameCommand::StartMarketingCampaign:
+                return "StartMarketingCampaign";
+            case GameCommand::PlaceMazeDesign:
+                return "PlaceMazeDesign";
+            case GameCommand::PlaceBanner:
+                return "PlaceBanner";
+            case GameCommand::RemoveBanner:
+                return "RemoveBanner";
+            case GameCommand::SetSceneryColour:
+                return "SetSceneryColour";
+            case GameCommand::SetWallColour:
+                return "SetWallColour";
+            case GameCommand::SetLargeSceneryColour:
+                return "SetLargeSceneryColour";
+            case GameCommand::SetBannerColour:
+                return "SetBannerColour";
+            case GameCommand::SetLandOwnership:
+                return "SetLandOwnership";
+            case GameCommand::ClearScenery:
+                return "ClearScenery";
+            case GameCommand::SetBannerName:
+                return "SetBannerName";
+            case GameCommand::SetSignName:
+                return "SetSignName";
+            case GameCommand::SetBannerStyle:
+                return "SetBannerStyle";
+            case GameCommand::SetSignStyle:
+                return "SetSignStyle";
+            case GameCommand::SetPlayerGroup:
+                return "SetPlayerGroup";
+            case GameCommand::ModifyGroups:
+                return "ModifyGroups";
+            case GameCommand::KickPlayer:
+                return "KickPlayer";
+            case GameCommand::Cheat:
+                return "Cheat";
+            case GameCommand::PickupGuest:
+                return "PickupGuest";
+            case GameCommand::PickupStaff:
+                return "PickupStaff";
+            case GameCommand::BalloonPress:
+                return "BalloonPress";
+            case GameCommand::ModifyTile:
+                return "ModifyTile";
+            case GameCommand::EditScenarioOptions:
+                return "EditScenarioOptions";
+            case GameCommand::PlacePeepSpawn:
+                return "PlacePeepSpawn";
+            case GameCommand::SetClimate:
+                return "SetClimate";
+            case GameCommand::SetColourScheme:
+                return "SetColourScheme";
+            case GameCommand::SetStaffCostume:
+                return "SetStaffCostume";
+            case GameCommand::PlaceFootpathAddition:
+                return "PlaceFootpathAddition";
+            case GameCommand::RemoveFootpathAddition:
+                return "RemoveFootpathAddition";
+            case GameCommand::GuestSetFlags:
+                return "GuestSetFlags";
+            case GameCommand::SetDate:
+                return "SetDate";
+            case GameCommand::Custom:
+                return "Custom";
```

Comment:
Shouldn't this be wrapped in `#ifdef ENABLE_SCRIPTING`?

May be nicer to refactor this entire function out, and have the names be part of the action classes, perhaps supplied where registered or as a template parameter. Probably better for another PR, though.

Reply:
The id exists regardless, so I dunno.